blog/posts/2021-02-21-serverless-functions-ember-glimmer.md

```diff
@@ -0,0 +1,212 @@
+---
+authorId: jaredgalanis
+categories:
+  - serverless
+  - ember
+  - glimmer
+  - javascript
+date: '2021-02-20'
+slug: serverless-functions-ember-glimmer
+title: Serverless Functions in Ember and Glimmer
+---
+
+I have recently spent some time looking at [API Routes](https://nextjs.org/docs/api-routes/introduction) in Next.js and I must say that I've been very happy with the value-add and developer experience / ergonomics they provide. 
+
+I really wanted to enjoy that experience in Ember and Glimmer and that is what we're covering in this post.
+
+## Table of Contents
+1. [Next API Routes](#nextapiroutes)
+2. [Netlify Dev](#netlifydev)
+3. [Netlify functions with Ember](#netlifyfunctionswithember)
+4. [Netlify functions with Glimmer](#netlifyfunctionswithglimmer)
+5. [Vercel Dev](#verceldev)
+6. [Vercel functions with Ember](#vercelfunctionswithember)
+7. [Vercel functions with Glimmer](#vercelfunctionswithglimmer)
+
+## Next API Routes
+
+If you are not familiar with Next's API routes, they are essentially node functions that deploy as serverless lambdas. These functions can be used in a lot of different ways: 
+
+- they could provide a place to write a GraphQL endpoint
+- they could serve as a gateway to other API's
+- they could serve as a full-fledged backend that talks to a persistence layer
+
+There a lot of possibilities here. API routes are especially helpful when performing operations that need to access secret environment variables that you would typically not want exposed to your client. For instance, if you need to generate presigned AWS s3 urls for the client to post files to.
+
+From a developer efficiency perspective, this could help unblock frontend developers on your team from waiting for backend developers to be available for certain tasks. Since API routes provide, well, an API, you could develop certain server side features alongside your frontend without much loss of context.
```

Comment:
```suggestion
From a developer efficiency perspective, this could help unblock front end developers on the team from waiting for backend developers to be available for certain tasks. Since API routes provide, well, an API, we can develop certain server side features alongside our frontend without much loss of context.
```